### ISSUE TITLE
composer autoload optimization

### ISSUE THREAD

**Issue by FoxxMD** [maciejczyzewski/bottomline#60]
Running composer 2.0.11 on PHP 7.2.34

When I run composer with [autoload optimization](https://getcomposer.org/doc/articles/autoloader-optimization.md) I get the following warning about bottomline:
```
Class __\__ located in ./vendor/maciejczyzewski/bottomline/src/__/load.php does not comply with psr-4 autoloading standard. Skipping.
```
Looking at bottomline's composer file it seems optimizing should work, but composer says otherwise. What can I do to fix this?

**Comment by allejo** [comment]
Looks to me like this is happening because our namespace is `__\\`. If I change the namespace to something else, Composer handles this just fine? I don't see anything in the [PSR-4 standard](https://www.php-fig.org/psr/psr-4/#2-specification) that would say a namespace of `__` is invalid. I'll investigate over the weekend (hopefully) to see what I can find.

When you generate an optimized autoloader, does your code cease to function?

**Comment by FoxxMD** [comment]
No code is still all good with optimized loader. I believe that if it is skipped it just means

> By default, the Composer autoloader runs relatively fast. However, due to the way PSR-4 and PSR-0 autoloading rules are set up, it needs to check the filesystem before resolving a classname conclusively.

Instead of doing that works upfront in the optimization at install time, once. Potentially on every request.

**Comment by allejo** [comment]
Less than a year later, fixed in 6fd205c1cb140292c3c21bfc6d2b567a14c7d7fb.
